caching-attribute-service-client/src/test/java/org/hypertrace/core/attribute/service/cachingclient/AttributeServiceCachedClientTest.java

```diff
@@ -0,0 +1,63 @@
+package org.hypertrace.core.attribute.service.cachingclient;
+
+import static java.util.Collections.emptyList;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import com.typesafe.config.ConfigFactory;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class AttributeServiceCachedClientTest extends CachingClientTestUtils {
```

Comment:
May be irrelevant with the ask to split the client modules, but in general we don't want tests extending other tests (and here particularly, it certainly fails the is-a rule). If you want to share some common utils, that's fine, but inheritance isn't the right tool for that.